### PR TITLE
Add seekToBeginning option, reuse instances of kafka (de)serializers

### LINKED ISSUE #90: Can I just provide a org.apache.kafka.common.serialization.Serializer instead of the library's one?
Hi!
I can not get the point in having ```monix.kafka.Serializer```. Can you explain me why do I need it and why do you use that reflection approach? Can I use Serializers and Deserializers from the ```org.apache.kafka.common``` to create ```KafkaConsumerObservable``` and ```KafkaProducerSink```?

### LINKED ISSUE #175: Add monix.kafka.KafkaConsumerConfig.observableSeekToBeginningOnStart
Currently we have ability to seek to the end of partitions (via specifying  `monix.kafka.KafkaConsumerConfig.observableSeekToEndOnStart` field). 

I think it would be nice to have ability to seek to the beginning also.

Any reasons why it wasn't implemented initially?

### PR DESCRIPTION
**Pull request by Avasil**
Fixes #175 
Fixes #90 

@yeryomenkom is that what you had in mind?

### CODE REVIEW

File: kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala [hunk 1]
```diff
@@ -249,6 +249,7 @@ final case class KafkaConsumerConfig(
   observableCommitType: ObservableCommitType,
   observableCommitOrder: ObservableCommitOrder,
   observableSeekToEndOnStart: Boolean,
+  observableSeekToBeginningOnStart: Boolean,
```
Comment (Avasil) [inline kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala:252]: Will probably change to ADT, I need to decide whether I want to break it or not

Comment (yeryomenkom) [reply kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala:252]: It would be nice to have this change before final release.

**Comment by yeryomenkom** [comment]
@Avasil Hi! Thank you for this PR. Seems good!

**Comment by Avasil** [comment]
Thanks @yeryomenkom - I will finish it and will do a release
ping me if I forget :D

**Pull request merged**